src/remarx/app/log_viewer.py

```diff
@@ -0,0 +1,91 @@
+"""Utilities for rendering remarx logs inside marimo notebooks."""
+
+from __future__ import annotations
+
+import pathlib
+from typing import Protocol, runtime_checkable
+
+import marimo as mo
+
+
+@runtime_checkable
+class _ReadablePath(Protocol):
+    """Protocol for any object that exposes a Path-like read_text method."""
+
+    name: str
+
+    def read_text(
+        self,
+        encoding: str | None = None,
+        errors: str | None = None,
+    ) -> str:  # pragma: no cover - Protocol
+        """Return the file contents as text."""
+
```

Comment:
Is this purely for typing? Could you not use pathlib or the `FileState` object that marimo returns?